### PR TITLE
feat: support detect huawei npu and deploy to it

### PR DESCRIPTION
**Pull request by aiwantaozi**

### CODE REVIEW

File: gpustack/worker/tools_manager.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def prepare_tools(self):
         self.download_fastfetch()
 
     def download_llama_box(self):
-        version = "v0.0.59-sp2"
+        version = "v0.0.59-sp3"
```
Comment (gitlawr) [inline gpustack/worker/tools_manager.py]: Use v0.0.64

File: gpustack/detectors/plugins/npu_smi/npu_smi.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    def gather_gpu_mapping(self) -> Dict[tuple[int], int]:
+        command = self._command_gather_gpu_mapping()
+        results = self._run_command(command)
+        mapping = self._decode_gpu_device_mapping(results)
+        return mapping
+
+    def _decode_gpu_devices(  # noqa: C901
+        self, result, gpu_mapping: Dict[tuple[int], int]
+    ) -> GPUDevicesInfo:
```
Comment (gitlawr) [inline gpustack/detectors/plugins/npu_smi/npu_smi.py]: Can we add unit tests for the parsing?
Add type hint to `result`

File: gpustack/detectors/detector_factory.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+            if device.vendor.lower() not in [
+                vendor.lower()
+                for vendor in [
+                    VendorEnum.NVIDIA,
+                    VendorEnum.Apple,
+                    VendorEnum.MTHREADS,
+                ]
+            ]:
+                # Ignore the device without memory.
```
Comment (gitlawr) [inline gpustack/detectors/detector_factory.py]: Move the comment to correct place

File: gpustack/detectors/detector_factory.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            if device.memory.total == 0:
+                continue
+
+            if device.vendor.lower() not in [
+                vendor.lower()
+                for vendor in [
+                    VendorEnum.NVIDIA,
+                    VendorEnum.Apple,
+                    VendorEnum.MTHREADS,
+                ]
```
Comment (gitlawr) [inline gpustack/detectors/detector_factory.py]: Where is Ascend?

File: gpustack/detectors/detector_factory.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+from gpustack.detectors.npu_smi.npu_smi import NPUSMI
+
+
+logger = logging.getLogger(__name__)
+
+
+class DetectorFactory:
+    def __init__(
+        self,
+        gpu_detectors: Dict[str, GPUDetector] = None,
```
Comment (gitlawr) [inline gpustack/detectors/detector_factory.py:18]: How do we use this param? I don't find reference

File: gpustack/worker/tools_manager.py [hunk 6]
```diff
@@ -64,7 +64,7 @@ def prepare_tools(self):
         self.download_fastfetch()
 
     def download_llama_box(self):
-        version = "v0.0.59-sp2"
+        version = "v0.0.64"
```
Comment (gitlawr) [inline gpustack/worker/tools_manager.py]: bump this

**Comment by gitlawr** [comment]
Can you update install.sh to only install extra dependencies on cuda environment

File: gpustack/detectors/detector_factory.py (hunk 5 above)
Comment (aiwantaozi) [reply gpustack/detectors/detector_factory.py:18]: This is for user-defined worker resources

File: install.sh [hunk 7]
```diff
@@ -405,9 +409,10 @@ install_gpustack() {
   verify_system
   install_dependencies
   check_python_tools
-  check_cuda
+  cuda_status=$(check_cuda)
```
Comment (gitlawr) [inline install.sh]: Define a `DEVICE` var and do it the same way as in `detect_os`

**Review by gitlawr — APPROVED** [review]

**Pull request merged**